src/backend/web.rs

```diff
@@ -55,19 +54,46 @@ pub(crate) struct RenderBundleEncoder(web_sys::GpuRenderBundleEncoder);
 // This is safe on wasm32 *for now*, but similarly to the unsafe Send impls for the handle type
 // wrappers, the full story for threading on wasm32 is still unfolding.
 
-pub(crate) struct MakeSendFuture<F>(F);
+pub(crate) struct MakeSendFuture<F, M> {
+    future: F,
+    map: Option<M>,
```

Comment:
We probably don't need to worry about the `map` being used multiple times, because the caller should stop polling once `Poll::Ready` happens anyway.  So maybe we could remove the `Option` here to just have `M`, and also remove the `assert`